### ISSUE TITLE
[0.17 beta] User configuration in jspm.config.js

### ISSUE THREAD

**Issue by rpkilby** [jspm/jspm-cli#1590]
We have a number of custom mappings and packages in our jspm config and we're running into the issue where installing/uninstalling packages can muck with the ordering. Our current solution is to do the following:

```js
SystemJS.config({
  map: {
    "assert": "github:jspm/nodelibs-assert@0.2.0-alpha",
    "buffer": "github:jspm/nodelibs-buffer@0.2.0-alpha",
    ...

    "zz": "",
    "custom": "mappings",
    "etc": "etc",
    ...
```

It works, but yeah. Is there a better way of doing this that we've missed? If not, it would be nice to have a clean way of separating out user configuration from managed configuration via the CLI. Maybe allow for multiple `SystemJS.config` calls in the file, and mark the managed config with a key? eg,

```js
SystemJS.config({
  managed: true,

  packageConfigPaths: [
    "npm:@*/*.json",
    "npm:*.json",
    "github:*/*.json"
  ],
  transpiler: "plugin-babel",
  ...
})

SystemJS.config({
  managed: false,
  map: {
    ...
  }
})
```

Thanks.

**Comment by guybedford** [comment]
This is a great point.

I've added an automatic implementation for this in https://github.com/jspm/jspm-cli/commit/e7e4e33c4a20ef503c303afa1ef9eee39ebc00f3 to split into managed configuration and manual user configuration parts. It definitely makes it more approachable this way.

**Comment by rpkilby** [comment]
:+1: many thanks

**Comment by guybedford** [comment]
Released in 0.17.0-beta.10.

**Comment by capaj** [comment]
@guybedford very good addition
